### PR TITLE
LVM-472 - Refactor Assessment

### PR DESCRIPTION
**Pull request by xingyif**
Modify assessment_helper to match official doc
https://cs5500-jira.ccs.neu.edu/browse/LVM-472

Add/rename fields for assessment
https://cs5500-jira.ccs.neu.edu/browse/LVM-473

View: Restrict input for assessment using drop down
https://cs5500-jira.ccs.neu.edu/browse/LVM-468

### CODE REVIEW

File: app/controllers/assessments_controller.rb [hunk 1]
```diff
@@ -50,6 +50,7 @@ def set_assessment
   end
 
   def assessment_params
-    params.require(:assessment).permit(:score, :date, :subject)
+    params.require(:assessment).permit(:score, :date, :category,
+                                       :level, :name, :assessment_type)
```
Comment (zack) [inline app/controllers/assessments_controller.rb]: `student_id` not necessary?

File: app/models/assessment.rb [hunk 2]
```diff
@@ -2,4 +2,9 @@ class Assessment < ApplicationRecord
   belongs_to :student
 
   validates :score, presence: true, length: { maximum: 6 }
+  validates :category, presence: true
+  validates :name, presence: true
+  validates :level, presence: true
+  validates :assessment_type, presence: true
+  validates :date, presence: true
```
Comment (zack) [inline app/models/assessment.rb:9]: we could also validate proper pairings between assessment categories, names, and levels. but that's a 'nice to have'.

File: app/views/assessments/show.html.erb [hunk 3]
```diff
@@ -1,4 +1,24 @@
 <p>
```
Comment (zack) [inline app/views/assessments/show.html.erb:1]: This page should probably have the student's name on it somewhere.

**Review by zack — CHANGES_REQUESTED** [review]
1) Can we please use the new table view style for the index page?

2) We need to have assessment indexes scoped to the student.

3) Coordinators should not be able to see other students' scores.

4) New assessments aren't in any way associated with a student.

File: app/views/assessments/show.html.erb (hunk 3 above)
Comment (xingyif) [reply app/views/assessments/show.html.erb:1]: added

**Review by zack — APPROVED** [review]

**Pull request merged**